src/express/utils/get-request-headers.ts

```diff
@@ -8,10 +8,10 @@ export function getRequestHeaders(
   const multiValueHeaders: Record<string, string[]> = {};
 
   for (const [key, value] of Object.entries(req.headers)) {
-    if (Array.isArray(value)) {
-      multiValueHeaders[key] = value;
-    } else if (typeof value === `string`) {
-      headers[key] = value;
+    if (value) {
+      const values = [value].flat();
+      multiValueHeaders[key] = values;
+      headers[key] = values[values.length - 1] || ``;
```

Comment:
The or case is unnecessary because you know the array has at least one value (see line 12). If TS has compiler errors, you could add a type to values e.g. `[unknown, ...unknown[]]`